include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -116,11 +116,28 @@ class TensorProductMatrixSymmetricSumBase
    * described in the main documentation of TensorProductMatrixSymmetricSum.
    * This function is operating on ArrayView to allow checks of
    * array bounds with respect to @p dst and @p src.
+   *
+   * @warning This function works on an internal temporal array, leading to
+   * increased memory consumption if many instances of this class is created,
+   * e.g., in the case of non-Cartesian meshes. Futhermore, only one thread run
```

Comment:
```suggestion
   * e.g., a different object on every cell with different underlying coefficients each. Furthermore, only one thread run
```
I think the term "non-Cartesian cells" cannot be understood without context, because there is no tensor product structure on those cells. Hence, I would just say the suggested text above.